src/Servers/Kestrel/Transport.Sockets/src/SocketTransportOptions.cs

```diff
@@ -166,5 +166,5 @@ public static Socket CreateDefaultBoundListenSocket(EndPoint endpoint)
         return listenSocket;
     }
 
-    internal Func<MemoryPool<byte>> MemoryPoolFactory { get; set; } = System.Buffers.PinnedBlockMemoryPoolFactory.Create;
+    internal IMemoryPoolFactory<byte> MemoryPoolFactory { get; set; } = DefaultSimpleMemoryPoolFactory.Instance;
```

Comment:
Are we at all concerned with breaking reflection-based code that was setting this previously?

Reply:
Not really. The couple teams we gave code to were told the private reflection wouldn't work in 10.0.